File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03013/p03013_num0_parsed.py
MOD = 10**9 + 7

def count_ways(n, m, broken):
    dp = [0] * (n + 1)
    dp[0] = 1

    for i in range(1, n + 1):
        if i in broken:
            dp[i] = 0
        else:
            if i - 1 >= 0:
                dp[i] += dp[i - 1]
            if i - 2 >= 0:
                dp[i] += dp[i - 2]
            dp[i] %= MOD

    return dp[n]

N, M = map(int, input().split())
a = [int(input()) for _ in range(M)]
print(count_ways(N, M, a))

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03013/p03013_num0_parsed.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int MOD = 1000000007;
    int N, M;
    cin >> N >> M;

    vector<long long> dp(N+1, 0);
    dp[0] = 1;
    vector<int> broken(M);
    for(int i = 0; i < M; i++) {
        cin >> broken[i];
    }

    for(int i = 1; i <= N; i++) {
        if(find(broken.begin(), broken.end(), i) == broken.end()) {
            if(i == 1) {
                dp[i] = dp[i-1] % MOD;
            } else {
                dp[i] = (dp[i-1] + dp[i-2]) % MOD;
            }
        }
    }

    cout << dp[N] << endl;
    return 0;
}